onnxruntime/test/python/onnxruntime_test_collective.py

```diff
@@ -223,44 +232,50 @@ def test_all_gather_axis1(self):
 
         np.testing.assert_allclose(outputs[0], expected_output, err_msg="all gather on axis1: result mismatch")
 
-    def test_all_to_all(self):
-        for np_elem_type, elem_type, communication_elem_type in (
+    @parameterized.expand(
+        [
             (np.float32, TensorProto.FLOAT, TensorProto.FLOAT),
             (np.int64, TensorProto.INT64, TensorProto.INT64),
-            # TODO: Fix the following case, which triggers random number-mismatch error.
-            # (np.float32, TensorProto.INT64, TensorProto.BOOL),
-        ):
-            model = self._create_alltoall_ut_model(
-                (128, 128), elem_type=elem_type, communication_elem_type=communication_elem_type
-            )
-            rank, size = self._get_rank_size()
-            ort_sess = ort.InferenceSession(
-                model.SerializeToString(),
-                providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
-                provider_options=[{"device_id": str(rank)}, {}],
-            )
+            (np.int64, TensorProto.INT64, TensorProto.BOOL),
+        ]
+    )
+    def test_all_to_all(self, np_elem_type, elem_type, communication_elem_type):
+        model = self._create_alltoall_ut_model(
+            (128, 128), elem_type=elem_type, communication_elem_type=communication_elem_type
+        )
+        rank, size = self._get_rank_size()
+        ort_sess = ort.InferenceSession(
+            model.SerializeToString(),
+            providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
+            provider_options=[{"device_id": str(rank)}, {}],
+        )
 
+        if communication_elem_type == TensorProto.BOOL:
+            input = np.ones((128, 128), dtype=np_elem_type) if rank == 0 else np.zeros((128, 128), dtype=np_elem_type)
```

Comment:
Please elaborate this computation with example. Thanks.

Reply:
Done!